pyo3-macros-backend/src/pyclass.rs

```diff
@@ -628,6 +759,201 @@ fn impl_enum(
     })
 }
 
+fn impl_complex_enum(
+    complex_enum: PyClassComplexEnum<'_>,
+    args: &PyClassArgs,
+    doc: PythonDoc,
+    methods_type: PyClassMethodsType,
+) -> Result<TokenStream> {
+    // Need to rig the enum PyClass options
+    let args = {
+        let mut rigged_args = args.clone();
+        // Needs to be frozen to disallow `&mut self` methods, which could break a runtime invariant
+        rigged_args.options.frozen = parse_quote!(frozen);
+        // Needs to be subclassable by the variant PyClasses
+        rigged_args.options.subclass = parse_quote!(subclass);
+        rigged_args
+    };
+
+    let krate = get_pyo3_crate(&args.options.krate);
+    let cls = complex_enum.ident;
+    let variants = complex_enum.variants;
+    let pytypeinfo = impl_pytypeinfo(cls, &args, None);
+
+    let default_slots = vec![];
```

Comment:
For the simple enums we generate a default `__repr__`, I wonder if that's useful here?

We also generate a default `__richcmp__` ordering conversion, which I guess we cannot do for this complex enum without opt-in for the user. Makes me more sure we want to have `#[pyclass(ord, eq)]` as opt-ins which simple enums could then opt in to, to make things more consistent. But that's a problem for a separate PR I think.

Reply:
I agree about `__richcmp__` being opt-in.

AFAICT generating a default `__repr__` for a complex enum would only be possible if all variant field types also have `__repr__`. I'm not sure if there are facilities to perform that check... So maybe this means `__repr__` should be opt-in as well?

Reply:
Yes, I think that `#[pyclass(repr)]` which relies on `Debug` impls would also make a lot of sense.